server/lib/parties/party-service.ts

```diff
@@ -234,8 +239,38 @@ export default class PartyService {
   }
 
   leaveParty(partyId: string, userId: number, clientId: string) {
+    const party = this.parties.get(partyId)
+    if (!party) {
```

Comment:
imo these checks present a security issue. I know it feels *nice* to just retrieve the party and check the members, but by doing things this way you're letting users verify the existence of parties they aren't in.

I'd prefer that you retrieve the user's current party from their client ID and check that it matches the party ID they passed, and the order of the errors would then be:
- You're not in a party
- You're not in *that* party

At no point would we tell them "that party exists but you're not in it"